iroh-dns-server/src/store.rs

```diff
@@ -0,0 +1,110 @@
+use std::path::Path;
+
+use anyhow::Result;
+use bytes::Bytes;
+use iroh_metrics::inc;
+use pkarr::{PublicKey, SignedPacket};
+use redb::{backends::InMemoryBackend, Database, ReadableTable, TableDefinition};
+
+use crate::metrics::Metrics;
+
+type PublicKeyBytes = [u8; 32];
+
+const SIGNED_PACKETS_TABLE: TableDefinition<&PublicKeyBytes, &[u8]> =
+    TableDefinition::new("signed-packets-1");
+
+pub struct SignedPacketStore {
+    db: Database,
+}
+
+impl SignedPacketStore {
+    pub fn open_file(path: impl AsRef<Path>) -> Result<Self> {
+        let db = Database::builder().create(path)?;
+        Self::open(db)
+    }
+
+    pub fn open_inmemory() -> Result<Self> {
+        let db = Database::builder().create_with_backend(InMemoryBackend::new())?;
+        Self::open(db)
+    }
+
+    pub fn open(db: Database) -> Result<Self> {
+        let write_tx = db.begin_write()?;
```

Comment:
This is called from async. I don't think it's a big deal, since while this will do IO it will be usually fast. But still, maybe it would be best to do the actor stuff... See #4